src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -0,0 +1,372 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmextractlabels.h"
+#include "qgsprocessingparameters.h"
+#include "qgsmapthemecollection.h"
+#include "qgsmaprenderercustompainterjob.h"
+#include "qgsnullpainterdevice.h"
+#include "qgslabelsink.h"
+#include "qgslayertree.h"
+#include "qgsvectorlayer.h"
+#include "qgsscalecalculator.h"
+#include "qgstextlabelfeature.h"
+#include "qgstextdocument.h"
+
+#include "pal/feature.h"
+#include "pal/pointset.h"
+#include "pal/labelposition.h"
+
+#include <QPainter>
+
+#include <cmath>
+
+///@cond PRIVATE
+
+QString QgsExtractLabelsAlgorithm::name() const
+{
+  return QStringLiteral( "extractlabels" );
+}
+
+QString QgsExtractLabelsAlgorithm::displayName() const
+{
+  return QObject::tr( "Extract labels" );
+}
+
+QStringList QgsExtractLabelsAlgorithm::tags() const
+{
+  return QObject::tr( "map themes" ).split( ',' );
+}
+
+QgsProcessingAlgorithm::Flags QgsExtractLabelsAlgorithm::flags() const
+{
+  return QgsProcessingAlgorithm::flags() | FlagRequiresProject;
+}
+
+QString QgsExtractLabelsAlgorithm::group() const
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsExtractLabelsAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsExtractLabelsAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterExtent(
+                  QStringLiteral( "EXTENT" ),
+                  QObject::tr( "Map extent" ) ) );
+
+  addParameter( new QgsProcessingParameterScale(
+                  QStringLiteral( "SCALE" ),
+                  QObject::tr( "Map scale" ) ) );
+
+  addParameter( new QgsProcessingParameterNumber(
+                  QStringLiteral( "DPI" ),
+                  QObject::tr( "Map resolution (in DPI)" ),
+                  QgsProcessingParameterNumber::Double,
+                  QVariant( 96.0 ) ) );
+
+  addParameter( new QgsProcessingParameterMapTheme(
+                  QStringLiteral( "MAP_THEME" ),
+                  QObject::tr( "Map theme" ),
+                  QVariant(), true ) );
+
+  addParameter( new QgsProcessingParameterFeatureSink(
+                  QStringLiteral( "OUTPUT" ),
+                  QObject::tr( "Extracted labels" ),
+                  QgsProcessing::TypeVectorPoint ) );
+}
+
+QString QgsExtractLabelsAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Extract label information (position, font details, etc.) from a rendered map at a given extent and scale." );
```

Comment:
How about something more concise here, like "Converts map labels to a point layer" ?